### DISCUSSION TITLE
Nix or Python installation

### DISCUSSION THREAD

**Discussion by engrvip123** [efabless/cace#118] [Q&A] [answered]
If all required tools are already custom-installed, is it advisable to use the Python-based installation method for CACE?  
Does using the Nix-based installation method again install the required tools, potentially using extra storage?

https://cace.readthedocs.io/en/latest/getting_started/installation_overview.html

**Comment by mole99** [comment, marked as answer]
Hi @engrvip123! You are correct in that the tools will be duplicated, requiring extra storage. So if you use the Python-based installation method and manage all the tools yourself, then there is no reason to also use the Nix-based installation method.
The advantage of the Nix-based installation method is that we can guarantee that you are running the correct versions of the tools that work well with CACE. We will make sure to update them regularly. If you have an issue with CACE, the first thing we would like to know is whether you are using the same version of the tools as in the Nix-based installation :)

**Reply by engrvip123** [reply]
Thanks @mole99 for clarification. Will CACE support only the latest versions of tools, or will it also be compatible with few earlier releases?
What kind of error/warnings are expected in CACE usage with incompatible tool versions?

Asking this to confirm that if a tool version causes an issue with CACE usage, then user can get to know about source of problem.

**Reply by mole99** [reply]
Officially, CACE supports the versions of the tools as listed in the Nix configuration. This way it is also the easiest for me to replicate any issues, as we are using exactly the same versions of the tools. Of course, CACE will probably also work if you use slightly different versions of the tools. But there may be changes to the tools' interfaces or supported options that could break CACE.

For example, xschem has recently been given the option to enable `top_is_subckt` and `lvs_format` separately. CACE makes use of this feature to improve the accuracy of the schematic simulations.
If you are using an earlier version of xschem that does not recognize these options, then CACE will probably fail during schematic netlist generation.

We are aware that CACE is also used in Docker containers such as [IIC-OSIC-TOOLS](https://github.com/iic-jku/iic-osic-tools), which bring their own tools. Therefore, we try not to rely on the very latest versions of the tools, but still use up-to-date, stable tool versions.
